mobile/Ihc/screens/NewTriageModal.js

```diff
@@ -0,0 +1,708 @@
+      ketone: 'Ketone',
+      bilirubin: 'Bilirubin',
+      unitswitch: false,
+    };
+  }
+
+  state = {
+    modalVisible: false,
+  };
+
+  setModalVisible(visible) {
+    this.setState({modalVisible: visible});
+  }
+
+  sexSwitch = (value) => {
+      //onValueChange of the switch this function will be called
+      this.setState({femaleswitch: value})
+      //state changes according to switch
+      //which will result in re-render the text
+   }
+
+   heightSwitch = (value) => {
+     this.setState({heightswitch: value})
+   }
+
+  weightSwitch = (value) => {
+    this.setState({weightswitch: value})
+  }
+
+  unitSwitch = (value) => {
+    this.setState({unitswitch: value})
+  }
+
+  tempSwitch = (value) => {
+    this.setState({tempswitch: value})
+  }
+
+  fastingSwitch = (value) => {
+    this.setState({fasting: value})
+  }
+
+  pregnantSwitch = (value) => {
+    this.setState({pregnant: value})
+  }
+
+  goToTriage = () => {
+    this.props.navigator.push({
+      screen: 'Ihc.TriageScreen',
+      title: 'Back to patient',
+      passProps: { name: this.props.name }
+    });
+  }
+
+  goToSoap = () => {
+    this.props.navigator.push({
+      screen: 'Ihc.SoapScreen',
+      title: 'Back to patient',
+      passProps: { name: this.props.name }
+    });
+  }
+
+  goToMedicationList = () => {
+    this.props.navigator.push({
+      screen: 'Ihc.MedicationScreen',
+      title: 'Back to patient',
+      passProps: { name: this.props.name }
+    });
+  }
+
+  goToHistory = () => {
+    this.props.navigator.push({
+      screen: 'Ihc.PatientHistoryScreen',
+      title: 'Back to patient',
+      passProps: { name: this.props.name }
+    });
+  }
+
+  goToGrowthChart = () => {
+    // Growth chart takes time to load
+    this.props.setLoading(true);
+    this.props.navigator.push({
+      screen: 'Ihc.GrowthChartScreen',
+      title: 'Back to patient',
+    });
+  }
+
+  onNavigatorEvent(event) {
+    if (event.id === 'willAppear') {
+      this.props.clearMessages();
+    }
+  }
+
+  render() {
+    const date = new Date();
+    //const dateString = `${date.getMonth()} ${date.getDate()}, ${date.getYear()}`;
+    const dateString = this.props.todayDateString || date.toDateString();
+    return (
+    <Container>
+
+    <View style={{marginTop: 22}}>
+        <Modal
+          animationType="slide"
+          transparent={false}
+          visible={this.state.modalVisible}
+          onRequestClose={() => {
+            Alert.alert('Modal has been closed.');
+          }}>
+          <ScrollView style={{marginTop: 22}}>
+            <View style={styles.triagescreen}>
+              <Text style={styles.title}>New Triage Form: {dateString}</Text>
+              <Text style={styles.title}>Patient Information</Text>
+
+              <View style={styles.triagesection}>
+                <Text style={styles.subtitle}>Name</Text>
+                <View style={styles.inputsection}>
+                <TextInput
```

Comment:
Is there a reason you are manually using TextInputs etc. rather than the same form library we are already using? example: https://github.com/TritonSE/ihc/blob/master/mobile/Ihc/screens/SigninScreen.js

The form library will make this code a _lot_ cleaner